crates/fullnode/src/runner.rs

```diff
@@ -325,19 +329,22 @@ where
         let accept_public_input_as_proven = self.accept_public_input_as_proven;
         let l1_block_cache = self.l1_block_cache.clone();
 
-        tokio::spawn(async move {
-            let l1_block_handler = L1BlockHandler::<C, Vm, Da, Stf::StateRoot, DB>::new(
-                ledger_db,
-                da_service,
-                sequencer_pub_key,
-                sequencer_da_pub_key,
-                prover_da_pub_key,
-                code_commitments_by_spec,
-                accept_public_input_as_proven,
-                l1_block_cache.clone(),
-            );
-            l1_block_handler.run(start_l1_height).await
-        });
+        self.task_manager
+            .spawn(move |cancellation_token| async move {
```

Comment:
if we can also pass the `cancellation_token` into l2 sync, we can make sure we committed db correctly before shutting down and that would be great